doc/developer/design/20230306_persist_mfp_pushdown.md

```diff
@@ -0,0 +1,272 @@
+non-null min and max values and null count. We may also track the min/max Row within the part.
+
+Statistics are optional: Persist may not be able to compute min/max values for all types, and we may need to trim statistics to keep metadata from getting too big.
+
+## Filtering
+
+Our compute layer pushes a `MapFilterProject` operator into the Persist source, which can filter and transform data before emitting it downstream.
+Before each batch part is fetched from Blob storage in `persist_source`, the source will use our new part statistics to analyze whether that MFP will filter out _all_ the rows in that part.
+If so, we get to skip fetching the part from Blob storage entirely.
+
+This requires that we have some way to figure out what an MFP's `MirScalarExpr`s might return given only basic information like the nullability and range of values. For more on how that can be done, see the section on [filtering parts].
+
+Not all parts may have statistics, and those statistics may not be complete.
+When stats are not present, this analysis needs to be conservative: if we don't have statistics for a column, we'll assume it might contain any legal value.
+
+## Filtering & Locality
+
+Having column statistics is a necessary building block to filter out parts before fetching them, but it is not guaranteed to be effective in actual queries unless there is some degree of locality to the filtered
+column values across parts in Blob storage.
+(If the data that matches our filter is evenly distributed across parts, we'll end up needing to fetch every part no matter how clever our analysis is!)
+
+In our first implementation, we plan to focus on _temporal filters_: ie. filters on columns that are correlated with wall-clock time.
+These filters tend to have very good locality, thanks to how Persist ingests and compacts data over time --
+the newest, and
+typically smallest, parts will contain the most recent updates, allowing one to filter out the older/larger
+parts and their data effectively. This is particularly effective for fetching recent data from append-only sources;
+we'll often have accumulated considerable numbers of older, larger parts that are no longer relevant to the filter.
+
+![mfp-temporal-filtering](https://user-images.githubusercontent.com/785446/224154154-2276a7d1-4149-407f-8737-9acd4e929396.png)
+
+# Reference explanation
+[reference-explanation]: #reference-explanation
+
+## Calculating statistics
+
+All statistics are stored in persist `State`, along with the other metadata about a batch and its parts. We might keep statistics at a part level, or a batch level, or (because it is possible to union the statistics as we have defined them below) both. In the future, we might explore keeping select statistics in `Blob` and storing only a pointer to them in State. This would allow us to keep much larger statistics, but may not end up being necessary.
```

Comment:
> We might keep statistics at a part level, or a batch level

Another newbie question: where can I refresh mu knowledge about the difference between a batch or a part? I _think_ that a part might consist of multiple batches, but I'm not sure. Also, I'm not sure whether all data within a part is guaranteed to be consolidated, or whether this applies only to the individual batches.